source/libip6string/stoip6.c

```diff
@@ -78,6 +78,37 @@ unsigned char  sipv6_prefixlength(const char *ip6addr)
     }
     return 0;
 }
+
+int stoip6_prefix(const char *ip6addr, void *dest, int16_t *prefix_len_out)
+{
+    size_t addr_len;
+    unsigned char prefix_length;
+
+    if (!ip6addr) {
+        return -1;
+    }
+
+    if (prefix_len_out) {
+        *prefix_len_out = -1;
+    }
+
+   addr_len = strlen(ip6addr);
+   prefix_length = sipv6_prefixlength(ip6addr);
+   if (prefix_length) {
```

Comment:
Hm, looks like sipv6_prefixlength is being bad. Need to distinguish "no prefix length" from "/0". Maybe pull in equivalent code inline, rather than call it? Would save strchr twice.